### PR TITLE
Fix refresh_course_metadata for subjects.

### PR DESCRIPTION
**Pull request by robrap**
The following fixes refresh_course_metadata for subjects.

### CODE REVIEW

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py [hunk 1]
```diff
@@ -141,6 +141,8 @@ def node_type(self):
 
     def process_node(self, data):
         slug = data['field_subject_url_slug']
+        # I have no idea if this is bogus, or what data will come in for language when not 'und'.
+        language_code = 'en-us' if (data['language'] == 'und') else data['language']
```
Comment (robrap) [inline course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: Reviewers: As noted, I only saw `'und'`, so not sure whether this really works.

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py [hunk 2]
```diff
@@ -149,19 +151,22 @@ def process_node(self, data):
             'card_image_url': self._get_nested_url(data.get('field_subject_card_image')),
             # NOTE (CCB): This is not a typo. Yes, the banner image for subjects is in a field with xseries in the name.
             'banner_image_url': self._get_nested_url(data.get('field_xseries_banner_image'))
-
         }
 
         # There is a bug with django-parler when using django's update_or_create() so we manually update or create.
         try:
             subject = Subject.objects.get(slug=slug, partner=self.partner)
+            subject.set_current_language(language_code)
```
Comment (robrap) [inline course_discovery/apps/course_metadata/data_loaders/marketing_site.py:161]: Reviewers: This seems to work if I manually create a subject with the same slug first.

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py [hunk 3]
```diff
@@ -149,19 +151,22 @@ def process_node(self, data):
...
             for key, value in defaults.items():
                 setattr(subject, key, value)
             subject.save()
         except Subject.DoesNotExist:
             new_values = {'slug': slug, 'partner': self.partner}
             new_values.update(defaults)
+            # In this case, this sets the language too late.  It already has an exception on the next line.
+            # > ValueError: language_code can't be null, use translation.activate(..) when accessing translated models outside the request/response loop.
             subject = Subject(**new_values)
```
Comment (robrap) [inline course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: Reviewers: Not sure how to make it happy regarding language with this type of object instantiation.

**Comment by robrap** [comment]
@bderusha: A reminder that it would be great to get some help resolving this at some point.  This is just a start.
There is no associated ticket, so we should add a bug and get into someone's backlog.  Thanks.
FYI: @mikedikan

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py (hunk 1 above)
Comment (bderusha) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: if data language is defined it will be a 2 letter code, so this default may be inconsistent with `data['language']` value

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py (hunk 3 above)
Comment (bderusha) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: looks like parlor uses `_current_language` as the model attribute.  Might work to set that as one of the `new_value` kv pairs

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py (hunk 1 above)
Comment (robrap) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: I'm not clear @bderusha because I would interpret "data language" and `data['language']` as the same thing, so I'm not sure what two things you are saying would be inconsistent.  Are you saying that `data['language']` would not be the same as a language_code, so we need to map from one to the other?  If so, can you clarify how?  If not, can you clarify what you are referring to?

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py (hunk 3 above)
Comment (robrap) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: I'll try that and see if I still get an exception.  Thanks.

**Review by robrap — COMMENTED** [review]
Thanks @bderusha.  See my responses.

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py (hunk 1 above)
Comment (bderusha) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: Ah sorry.  Let me try again.

Your default `language_code` here is `en-us` however if there is English content is specified in the `data['language']` field it would be the 2 letter code `en`.  This could result in inconsistent results for English content, some saved at `en` some as `en-us`

Comment (robrap) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: I see.  I think you are saying, change:
```
language_code = 'en-us'
```
to
```
language_code = 'en'
```

Comment (bderusha) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py]: Yes, that would be consistent with the language data we would expect here

**Comment by robrap** [comment]
@jaebradley: It was silly of me to not get this over the line when we were working on subjects.  I might need help with this.

I'm having trouble getting a subject to be pulled in when using `./manage.py refresh_course_metada` now.  I tried creating a new subject in edx-mktg, but probably did something wrong.  So, I can't see errors on master and I can't see if my PR fixes the issue. :(

**Comment by robrap** [comment]
@bderusha: This has been sitting around for a long time, and seemed like a waste, so I was trying to finally complete it.  However, looks like I have a little code coverage issue.  Can we overlook this? :)
Let me know if you think I can squash and merge.  Thanks.

**Comment by robrap** [comment]
@bderusha: Please ignore my last comment for now.  It was late and I really just wanted to merge this PR.  I'll look into unit tests.

**Comment by robrap** [comment]
@bderusha: This is now ready for re-review when you get the chance.  Thank you.

**Review by bderusha — APPROVED** [review]
Looks good.  rebase and squash and ship!
